### PR TITLE
DB-10383 Dropping an expression-based index also drops its statistics

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by jyuanca — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/AbstractDropIndexConstantOperation.java [hunk 1]
```diff
@@ -126,6 +123,7 @@ private void drop(ConglomerateDescriptor cd,
         SpliceTransactionManager userTxnManager = (SpliceTransactionManager)lcc.getTransactionExecute();
         dd.dropConglomerateDescriptor(cd,userTxnManager);
         td.removeConglomerateDescriptor(cd);
+        dd.deletePartitionStatistics(cd.getConglomerateNumber(), userTxnManager);
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/AbstractDropIndexConstantOperation.java:126]: We may need to broadcast the ALTER_STATS messages to invalidate the plans based on that stats too and remove the entry from the dictionary cache. See StatisticsAdmin.ddlNotification().

**Review by yxia92 — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/AbstractDropIndexConstantOperation.java (hunk 1 above)
Comment (ascend1) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/AbstractDropIndexConstantOperation.java:126]: Good to know! Added in the next commit.

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by yxia92 — APPROVED** [review]
+1

**Review by carolp-503 — APPROVED** [review]
+1

**Review by hatyo — APPROVED** [review]

**Review by dgomezferro — APPROVED** [review]

**Pull request merged**